lailib/torch/parameter_store.py

```diff
@@ -0,0 +1,110 @@
+import os
+import torch
+import numpy as np
+from os import listdir
+
+#TODO add support for optimizer tensor type
+#TODO add types for function heads
+#TODO support early stopping
+def save_network(network,
+                 optimizer,
+                 save_dir,
+                 model_name,
+                 global_step,
+                 use_gpu=True):
+    '''
+    save current neural network parameters and optimizer parameters
+    currently the use_gpu flag only takes care of parameters in network,
+    some optimizers has tensors in it. Be careful on optimizers when use
+    the save load functions
+
+    :param network: the pytorch neural network that should be saved
+    :param optimizer: the optimizer for network training
+    :param save_dir: path to the checkpoint directory
+    :param model_name: name of the model
+    :param global_step: An integer indicates how many steps the training
+                        had run, (usually steps is the number of iterations and
+                        number of epoches)
```

Comment:
string has more freedom, can add tag or disc as an optional parameter with default "" that comes before global step and still compare using global step